### DISCUSSION TITLE
Error while saving a row including column type JSON

### DISCUSSION THREAD

**Discussion by juangbm** [Budibase/budibase#9624] [Help]
Hi, 
I have a form to update a tabe, that includes a JSON field, but whenever I try to update (SAVE) I get an error about **Unknown column 'default_lang' in 'field list'**, this is how my form looks like

![image](https://user-images.githubusercontent.com/22911293/217825433-19ae685e-13e6-49e7-9f40-cd6a8bbf9755.png)

As you can see the information coming form the database is displayed correctly, however when trying to save the error appears.

Any idea what can be going on?

Thanks!

**Comment by andz-bb** [comment]
Hey @juangbm - could you send me an app export so that I can take a closer look at this for you on my end? You can send it to support@budibase.com or you can send it [on our Discord server here.](https://discord.gg/ZepTmGbtfF) Thanks!

**Reply by juangbm** [reply]
@andz-bb did you find anything about this issue? Is this a bug? I need to consider if we need to change the data type to Text instead of JSON if the JSON field save/update is not working

**Reply by andz-bb** [reply]
hey @juangbm - sorry for the late response! While we have full support for [JSON columns with the internal Budibase DB](https://docs.budibase.com/docs/json), at the minute we don't fully support that type from external datasources. I see in your app export you're using a JSON type column from a MySQL database, so this is the source of the problem in this case.

This has been raised as an issue here #3382 and has more recently been discussed as part of an issue with Budibase dealing with multi-select columns in external datasources (#8389). I think the safest option here would be to go for the text type if you can for now.

**Comment by juangbm** [comment]
Hi @andz-bb 
App export sent

Thanks!

**Reply by juangbm** [reply]
Hi @andz-bb, could you check the export? did you find out why it is not saving correctly?
